### PR TITLE
Min password length

### PR DESCRIPTION
**Pull request by jamesgeddes**
Minimum length requirement done. 

Need to recaptcha. Will split into its own issue.

### CODE REVIEW

**Comment by CarwynNelson** [comment]
It looks like Django has this built in and might be slightly more robust https://docs.djangoproject.com/en/3.0/topics/auth/passwords/#password-validation

Also, whilst 16 characters would be ideal there are many folks who might find remembering creating and remembering a password that long quite hard.

We could (and should) push people towards password managers, but maybe we should mention a few options instead of just lastpass? Maybe some FOSS options that aren't tied to a proprietary service?

**Comment by jamesgeddes** [comment]
We need to avoid complexity rules, so should just stick to length requirements. Always in favour of FOSS, but Lastpass will pay GZ for every paid account we refer.

**Comment by jamesgeddes** [comment]
Have not looked at the automated testing. Will have a look.

**Comment by jamesgeddes** [comment]
Password now requires min of 10 chars, just because its an easy round number. Once we release, we can put a link to the GZ password manager blog post in the registration form, but it's not obligatory for MVP.

**Comment by jamesgeddes** [comment]
updated automated tests with a permissible password. Should we do a test that checks to see if bad passwords are rejected?

**Comment by CarwynNelson** [comment]
> Should we do a test that checks to see if bad passwords are rejected?

Absolutely :)

**Review by CarwynNelson — CHANGES_REQUESTED** [review]
Could we add that extra test for passwords that are too short, plus a few small changes that I have noted in the review.

Overall this looks great :)

**Comment by CarwynNelson** [comment]
Also this test is now failing:
```
FAIL: test_signed_in_users_cannot_register (memberships.tests.test_registration_form.RegisterFormTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/project/memberships/tests/test_registration_form.py", line 26, in test_signed_in_users_cannot_register
    self.assertEqual(response.status_code, 403)
AssertionError: 200 != 403
```

File: memberships/forms.py [hunk 1]
```diff
@@ -33,3 +37,11 @@ def clean_birth_date(self, *args, **kwargs):
         # FIXME JDG in the future, we should put messages like these in the admin area for future changes
 
         return birth_date
+
+    def clean_password(self):
+        password = self.cleaned_data.get('password')
+        password_validation.validate_password(password, None)
+        # raise forms.ValidationError(mark_safe('Use a <a href="http://example.com" target="_blank">'
```
[the remark this answers is not in the document]
Comment (jamesgeddes) [reply memberships/forms.py:44]: Oops! Yes, on it now

File: web/settings.py [hunk 2]
```diff
@@ -91,8 +91,17 @@
         "NAME": (
             "django.contrib.auth.password_validation.UserAttributeSimilarityValidator"
         ),
+        "OPTIONS": {
+            "user_attributes": (
```
[the remark this answers is not in the document]
Comment (jamesgeddes) [reply web/settings.py:95]: specifies which elements should be checked for similarity

**Comment by jamesgeddes** [comment]
> Also this test is now failing:
> 
> ```
> FAIL: test_signed_in_users_cannot_register (memberships.tests.test_registration_form.RegisterFormTestCase)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/circleci/project/memberships/tests/test_registration_form.py", line 26, in test_signed_in_users_cannot_register
>     self.assertEqual(response.status_code, 403)
> AssertionError: 200 != 403
> ```

I have updated the password in `test_signed_in_users_cannot_register` so that it provides a permissible password.

File: web/settings.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (CarwynNelson) [reply web/settings.py:95]: It seems to me that all of these fields are likely to be similar. What does a value of 0.5 mean?

**Review by CarwynNelson — APPROVED** [review]

**Pull request merged**